### ISSUE TITLE
Confusing "Figure already exists" error message with 404 scenario

### ISSUE THREAD

**Issue by daka1510** [Qiskit/qiskit-ibm-experiment#22]
@mriedem commented on [Mon Feb 28 2022](https://github.com/Qiskit/qiskit-ibm-provider/issues/245)

### Information

- **Qiskit IBM Provider version**: latest (actually qiskit-ibmq-provider 0.18.3)
- **Python version**: n/a
- **Operating system**: n/a

### What is the current behavior?

qiskit-experiments parallel experiments first try to create the parent experiment and then the children experiments and upload plots (figures) for the child experiments. If the parent experiment fails to create then the children experiments will also fail (this seems like a separate issue where there are cascading failures that don't stop the processing), and then trying to create figures for those child experiments will fail with something like this:

```
Unable to save the experiment data: Traceback (most recent call last):
  File "/nas/systems/pok/pok008_system1/2021_03_24_F518_Bangkok/playing/mattias/qiskit-experiments/qiskit_experiments/database_service/utils.py", line 152, in save_data
    return True, new_func(**kwargs)
  File "/home/labuser/.virtualenvs/controller/lib64/python3.8/site-packages/qiskit/providers/ibmq/experiment/ibm_experiment_service.py", line 1104, in create_figure
    response = self._api_client.experiment_plot_upload(experiment_id, figure, figure_name,
  File "/usr/lib64/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/labuser/.virtualenvs/controller/lib64/python3.8/site-packages/qiskit/providers/ibmq/experiment/utils.py", line 32, in map_api_error
    raise IBMExperimentEntryNotFound(error_msg + f" {api_err}") from None
qiskit.providers.ibmq.experiment.exceptions.IBMExperimentEntryNotFound: 'Figure figure_47e90a4d_2022-02-21T09:57:47.352206_0.svg already exists. \'404 Client Error: Not Found for url: https://api.quantum-computing.ibm.com/resultsdb/experiments/47e90a4d-b3dd-4964-9345-4884cebd8382/plots. {"errors":["Experiment 47e90a4d-b3dd-4964-9345-4884cebd8382 does not exist"]}\''
```

Note that this is actually using qiskit-ibmq-provider since qiskit-ibm-provider isn't released yet, but it's the same code.

The bug is that the error message says `Figure figure_47e90a4d_2022-02-21T09:57:47.352206_0.svg already exists` which is misleading. It's hard-coded here:

https://github.com/Qiskit/qiskit-ibm-provider/blob/f4ff3b0dcf04ecb6cb21aea4e3045f312f77bbff/qiskit_ibm/experiment/ibm_experiment_service.py#L1121

The actual issue is the 404 on the experiment ID in the path - the experiment doesn't exist so you can't upload a plot for it. The "figure already exists" is confusing since it's a 404 rather than a 409 case.

### Steps to reproduce the problem

Try to upload a figure for an experiment which doesn't exist.

### What is the expected behavior?

I'd expect to see something about the experiment not existing - which is part of the error message, just not the first part.

### Suggested solutions

Maybe generalize the string to `map_api_error` to be like `f"Creating figure {figure_name} failed."`.

**Comment by daka1510** [comment]
Moving this issue from @mriedem (raised in `qiskit-ibm-provider`) because the experiment service will be implemented in `qiskit-ibm-experiment`.

@gadial - can you take a look?

**Comment by gadial** [comment]
This is a serious issue which I also encountered today independently and occurs in many places in the code; will try to fix this asap.

**Comment by gadial** [comment]
Fixed in #25
